src/volta-migrate.rs

```diff
@@ -1,11 +1,42 @@
+use structopt::StructOpt;
+
 use volta_core::error::{report_error, ExitCode};
+use volta_core::layout::volta_home;
 use volta_core::log::{LogContext, LogVerbosity, Logger};
 use volta_migrate::run_migration;
 
+#[derive(StructOpt)]
+#[structopt(
+    name = "volta-migrate",
+    about = "Migrates the Volta directory to the latest version",
+    raw(global_setting = "structopt::clap::AppSettings::ColoredHelp"),
+    raw(global_setting = "structopt::clap::AppSettings::ColorAuto"),
+    raw(global_setting = "structopt::clap::AppSettings::DeriveDisplayOrder"),
+    raw(global_setting = "structopt::clap::AppSettings::DisableVersion")
+)]
+struct VoltaMigrate {
```

Comment:
Can you expand a bit on the use-case behind this option? The migration is designed to be idempotent, so running it multiple times shouldn't be a concern. Additionally, Volta will absolutely not work without the directory set up (and will force it to be set up when it's run), so I'm not clear what the benefits would of calling `volta-migrate` and having it no-op would be.

Finally, while I agree that calling `volta-migrate` from the Homebrew installer makes sense, I think that `volta-migrate` (similar to `volta-shim`) is primarily intended to be an implementation detail, so I'm a little wary of adding StructOpt to it as that involves a lot of extra compilation baggage and makes it seem more like a CLI that people should think about.

Reply:
I think `brew intall volta` alone should not modify anything under the home directory, so `volta-migrate` have to be no-op unless  `~/.volta` exists. Not creating `~/.volta` up front is not a problem as a first-time user will probably run `volta install node` afterwards anyway.

On the other hand, `brew upgrade volta` should take care of existing shims so that the user doesn't have to run `volta setup` themselves again.

As for option parsing, how about just doing `std::env::args().nth(1) == Some("--no-create".to_owned())`?

Reply:
~~`std::env::args().nth(1) == Some("--no-create".to_owned())`~~
`std::env::args().nth(1).map_or(false, |v| v  == "--no-create")`

Reply:
It's a bit of a weird state, since the script installer _does_ modify the home directory (and even modifies the profile scripts unless turned off). However, it looks like the current Homebrew installer _doesn't_ do that (meaning users will need to run `volta setup` after `brew install volta`), and we probably shouldn't change that out from underneath them.

I think having a small flag is a reasonable, though I would prefer to do a more simplistic matching like you showed, as opposed to bringing in all the complexity of StructOpt. I think your solution is pretty good, though I would use `args_os` since that won't panic if there are non-unicode characters (and you can probably take advantage of the fact that `OsString` implements `PartialEq<&str>`):

```
matches!(std::env::args_os().nth(1), Some(flag) if flag == "--no-create")
```

We should also probably have a comment around that branch about how it's used by the Homebrew formula to avoid making changes to the user's system.

Reply:
> `matches!(std::env::args_os().nth(1), Some(flag) if flag == "--no-create")`

Didn't know that this is even possible! I replaced StructOpt with `matches!` and added a comment about the purpose of the flag.

Will squash & rebase If there is nothing else to discuss.